migrations/20231210123554_init.sql

```diff
@@ -0,0 +1,323 @@
+);
+
+
+-- EVENT ENTITIES
+
+CREATE TABLE event
+(
+    id             UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    -------------------------------------------------------
+    name           VARCHAR(255) NOT NULL,
+    description    TEXT,
+    website        VARCHAR(255),
+    accepts_staff  BOOLEAN NOT NULL,
+    start_date     DATE NOT NULL,
+    end_date       DATE NOT NULL,
+    avatar_url     VARCHAR(255),
+    -------------------------------------------------------
+    created_at     TIMESTAMP NOT NULL DEFAULT now(),
+    edited_at      TIMESTAMP NOT NULL DEFAULT now(),
+    deleted_at     TIMESTAMP,
+    -------------------------------------------------------
+    CONSTRAINT check_event_name_len
+        CHECK (char_length(name) >= 1),
+    CONSTRAINT check_event_start_date_lte_end_date
+        CHECK (start_date >= end_date),
+    CONSTRAINT check_event_created_at_lte_edited_at
+        CHECK (edited_at >= created_at)
+);
+
+
+CREATE TABLE associated_company
+(
+    type        association NOT NULL,
+    -------------------------------------------------------
+    created_at  TIMESTAMP NOT NULL DEFAULT now(),
+    edited_at   TIMESTAMP NOT NULL DEFAULT now(),
+    deleted_at  TIMESTAMP,
+    -------------------------------------------------------
+    company_id  UUID NOT NULL,
+    event_id    UUID NOT NULL,
+    -------------------------------------------------------
+    PRIMARY KEY (company_id, event_id),
+    FOREIGN KEY (company_id) REFERENCES company (id),
+    FOREIGN KEY (event_id) REFERENCES event (id),
+    -------------------------------------------------------
+    CONSTRAINT check_associated_company_created_at_lte_edited_at
+        CHECK (edited_at >= created_at)
+);
+
+
+CREATE TABLE timesheet
+(
+    id           UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    -------------------------------------------------------
+    start_date   DATE NOT NULL,
+    end_date     DATE NOT NULL,
+    total_hours  hours_per_month_float NOT NULL,
+    is_editable  BOOLEAN NOT NULL,
+    manager_note TEXT,
+    -------------------------------------------------------
+    created_at   TIMESTAMP NOT NULL DEFAULT now(),
+    edited_at    TIMESTAMP NOT NULL DEFAULT now(),
+    deleted_at   TIMESTAMP,
+    --------------------------------------------------------
+    user_id      UUID NOT NULL,
+    company_id   UUID NOT NULL,
+    event_id     UUID NOT NULL,
+    --------------------------------------------------------
+    FOREIGN KEY  (user_id) REFERENCES user_record (id),
+    FOREIGN KEY  (company_id) REFERENCES company (id),
+    FOREIGN KEY  (event_id) REFERENCES event (id),
+    --------------------------------------------------------
+    CONSTRAINT check_timesheet_start_date_lte_end_date
+        CHECK (start_date >= end_date),
+    CONSTRAINT check_timesheet_created_at_lte_edited_at
+        CHECK (edited_at >= created_at)
+);
+
+
+CREATE TABLE work_day
+(   
+    id           UUID PRIMARY KEY DEFAULT gen_random_uuid(),
```

Comment:
We identify work_day as (timesheet_id, date) as there should not be two work_day entries for the same day for one timesheet. Please, stick to the ERD model.